launcher.py

```diff
@@ -0,0 +1,498 @@
+            packInfo = json.load(packInfoFile)
+            packInfoFile.close()
+        
+        functionExecutor(delete, argsParser(splitArgs[0], addArgs = [instance, "delete", "return"]))
+        
+        zipFile = DOWNLOADPATH + f"{instance}.zip"
+        resp = download(packInfo['downloadUrl'], zipFile)
+        if resp != 200:
+            quit(Format.format(f"Error! Can't download {zipFile} from '{url}': {resp}", Format.RED))
+        
+        with open(DOWNLOADPATH + ".packinfo", "wt") as packInfoFile:
+            packInfoFile.writelines(json.dumps(packInfo))
+            packInfoFile.close()
+        
+        functionExecutor(install, argsParser(splitArgs[0], addArgs = ["local", "zip", zipFile]))
+    elif splitArgs.__contains__("reset") and splitArgs.__contains__("toDownload"):
+        functionExecutor(delete, argsParser(splitArgs[0], addArgs = [instance, "delete", "return"]))
+        functionExecutor(install, argsParser(splitArgs[0], addArgs = "download"))
+
+def cleanUp(returnToMenu = True, fullCleanUp = False):
+    print("Cleanup...")
+    if CLEANUPFILES:
+        try:
+            shutil.rmtree(DOWNLOADPATH)
+            if not fullCleanUp:
+                os.mkdir(DOWNLOADPATH)
+        except:
+            pass
+    if returnToMenu == False:
+        return
+    main("Home%install")
+
+################################################################################################################################################
+# this is the main function to manage the navigation
+# separator constants:
+PATHSEPARATOR = "%" # is used to get navigation path for example to serve the "back"-function
+ARGSSEPARATOR = "|" # is used to seperate arguments for the functions. Could have used a dict or list instead of a string with separators, but oh well...
+VARSPARSER = "&" # is used to signal an variable for MenuHelper (variable exchanger in Format.exchange())
+
+# file save constants:
+DOWNLOADPATH = os.path.dirname(__file__) + f"/.tmp/{time.time()}/" # path to temporarily download files
+PACKSPATH = os.path.dirname(__file__) + "/packs/"
+NEWINSTANCENAME = "&DisplayModpackVersion&" # available variables: ModpackName, ModpackId, DisplayModpackVersion, NowTime (Format: hhmmss), NowDateUS (Format: MMDDYY), NowDate (Format: DDMMYY) 
+# TODO: more arguments (around line: 288)
+# The Instance-Name gets transfered, if modpack is reset. Therefore if you use date and time vars it will NOT get updated (could be implemented, important credentials could be saved in .packinfo)
+
+# download constants
+BASEURL = "https://addons-ecs.forgesvc.net/api/v2/addon/" # curseforge API
+AGENT = "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36"
+HEADER = {"User-Agent": AGENT}
+
+# launch constants
+INSTALLERCOMMAND = f'python3 {os.path.dirname(__file__)}/install.py "&ZipFilePath&" --manual' # command to call installer.py
```

Comment:
`--manual` should be specified by the user (i.e. on `sys.argv`) and only if necessary. Forge 1.12 and Fabric installers currently work automatically, and I am planning on automating Forge installation for other versions.

Also, since we're in a Python script, I'd recommend actually importing `install.py` and running its `main()` function directly:
```python
import install as install_script
# instead of using subprocess, run this (zipfile is the path to the zip file, and manual is a boolean equivalent to specifying --manual
install_script.main(zipfile, manual)
```
This is how the other scripts in the main program interact, since they were originally standalone scripts during initial development.